tests/ui/let_underscore_future.stderr

```diff
@@ -8,28 +8,20 @@ LL |     let _ = some_async_fn();
    = note: `-D clippy::let-underscore-future` implied by `-D warnings`
 
 error: non-binding `let` on a future
-  --> $DIR/let_underscore_future.rs:15:5
+  --> $DIR/let_underscore_future.rs:17:5
    |
 LL |     let _ = custom();
    |     ^^^^^^^^^^^^^^^^^
    |
    = help: consider awaiting the future or dropping explicitly with `std::mem::drop`
 
 error: non-binding `let` on a future
-  --> $DIR/let_underscore_future.rs:19:5
+  --> $DIR/let_underscore_future.rs:21:5
    |
 LL |     let _ = future;
    |     ^^^^^^^^^^^^^^^
    |
    = help: consider awaiting the future or dropping explicitly with `std::mem::drop`
 
-error: this argument is a mutable reference, but not used mutably
```

Comment:
Wonder if we can have a CI test to prevent this stuff from happening 🤔

Reply:
We can probably search for `-D` followed by anything that isn't a specific lint. Not sure how you'd set that "specific lint" for each test though